### DISCUSSION TITLE
Migrate from MainfluxLabs

### DISCUSSION THREAD

**Discussion by voler88** [absmach/magistrala#2394] [Q&A] [answered]
Which version of Magistrala can I upgrade to from MainfluxLabs 0.16.3 without losing data (things, messages, certificates) or is it impossible?

**Comment by dborovcanin** [comment, marked as answer]
Hello, @voler88, and sorry for the late reply. MainfluxLabs is a separate entity that has no connection to Magistrala. They share the same Mainflux origin, but unfortunately, I don't think you can easily migrate because projects diverged significantly over the past few years.
